File: easy/python/0118.py
class Solution:
    def generate(self, numRows: 'int') -> 'List[List[int]]':
        if not numRows:
            return []
        row = [[1]]
        while numRows > 1:
            last = row[-1]
            new = []
            for i in range(1, len(last)):
                new += [last[i] + last[i-1]]
            new = [1] + new + [1]
            row.append(new)
            numRows -= 1
        return row
        

File: easy/cpp/0118.cpp
class Solution {
public:
    vector<vector<int>> generate(int numRows) {
        vector<vector<int>> res;
        if(!numRows) return res;
        vector<int> start{1};
        res.push_back(start);
        while(numRows-- > 1){
            vector<int> next{1};
            vector<int> last = res.back();
            for(int i = 1 ; i < last.size() ; i++){
                next.push_back(last[i] + last[i-1]);
            }
            next.push_back(1);
            res.push_back(next);
        }
        return res;
    }
};